src/installer/tests/HostActivation.Tests/NativeHosting/Ijwhost.cs

```diff
@@ -63,14 +109,28 @@ public void ManagedHost(bool selfContained)
 
         public class SharedTestState : SharedTestStateBase
         {
+            public string HostFxrPath { get; }
             public string IjwLibraryPath { get; }
-
+            public string IjwLibraryRuntimeConfigPath { get; }
             public TestProjectFixture ManagedHostFixture_FrameworkDependent { get; }
             public TestProjectFixture ManagedHostFixture_SelfContained { get; }
 
             public SharedTestState()
+                :this(null)
+            {}
+
+            private SharedTestState(SharedTestState other = null)
             {
-                string folder = Path.Combine(BaseDirectory, "ijw");
+                var dotNet = new Microsoft.DotNet.Cli.Build.DotNetCli(RepoDirectories.BuiltDotnet);
+                HostFxrPath = dotNet.GreatestVersionHostFxrFilePath;
+
+                string folder = Path.Combine(BaseDirectory, other==null?"ijw":"ijw_copy");
+
+                // make sure we start with a clean slate
+                if (Directory.Exists(folder))
+                {
+                    Directory.Delete(folder);
+                }
                 Directory.CreateDirectory(folder);
```

Comment:
Instead of making the entire `SharedTestState` copyable, can you make the folder containing all the ijw assets a `TestApp` (which already has a `Copy` function) and expose that as a property.

```suggestion
                string folder = Path.Combine(BaseDirectory, "ijw");
                IjwApp = new TestApp(folder, "ijw");
```

Then, you should be be able to do things like:

```c#
using (var app = sharedState.IjwApp.Copy())
{
    // app.AppDll instead of IjwLibraryPath
    // app.RuntimeConfigPath instead of IjwLibraryRuntimeConfigPath
    // app.Location instead of Path.GetDirectoryName(...)
}
```